src/Tests/SqlServer/QueryIntegrationTests.fs

```diff
@@ -658,4 +659,78 @@ let tests =
         }
 #endif
 
+        testTask "Insert, update, and select with both datetime and datetime2 precision" {
+            use ctx = openContext ()
+            ctx.BeginTransaction()
+            
+            let baseTimestamp = System.DateTime(2022,07,22, 11,50,28)
+            let timestamp = System.DateTime(baseTimestamp.Ticks + 1234567L)
+            // Simple insert of one entity
+            let entity': ProviderDbTypeTest.Test =
+                {
+                    ID = 0
+                    LessPrecision = timestamp
+                    MorePrecision = timestamp
+                }
+
+            let! _ = 
+                insert {
+                    into providerDbTypeTestTable 
+                    entity entity'
+                }
+                |> ctx.InsertAsync
+
+            let! retrievedBack = 
+                selectTask HydraReader.Read (Shared ctx) {
+                    for row in providerDbTypeTestTable do
+                    select row
+                }
+
+            Expect.equal [timestamp] [for (row:ProviderDbTypeTest.Test) in retrievedBack -> row.MorePrecision] "INSERT: Expected DATETIME2 to be stored with full precision"
+            Expect.notEqual [timestamp] [for (row:ProviderDbTypeTest.Test) in retrievedBack -> row.LessPrecision] "INSERT: Expected a loss of precision when storing a DATETIME"
+
+            let! fullPrecisionQuery = 
+                selectTask HydraReader.Read (Shared ctx) { 
+                    for row in providerDbTypeTestTable do
+                    where (row.MorePrecision = timestamp)
+                    count
+                }
+
+            let! lessPrecisionQuery = 
+                selectTask HydraReader.Read (Shared ctx) { 
+                    for row in providerDbTypeTestTable do
+                    where (row.LessPrecision = timestamp)
+                    count
+                }
+
+            Expect.equal fullPrecisionQuery 1 "SELECT: Expected precision of a DATETIME2 query parameter to match the precision in the database"
+            Expect.equal lessPrecisionQuery 1 "SELECT: Expected precision of a DATETIME query parameter to match the precision in the database"
```

Comment:
This fails without the changes to LinqExpressionVisitors.  `fullPrecisionQuery` is 0 because the timestamp in the where clause is truncated to DATETIME precision.